python/Scripts/definitionMaterials.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+'''
+Create a document per node definition such that an instance of that 
+definition is routed to a "material" root. Makes use of "convert"
+nodes for various output types. A material root is created for each
+output on a nodedef, if multiple exist.
+'''
```

Comment:
Could this be considered a test script that can stay local and not be included in the main public repo? Or is it generally useful to be included here? Just a thought.

Reply:
I think generating these graphs creates a smaller subset of graphs for test than what was hand-done.
So was thinking that this could be done as part of the test setup. 
Also maybe it can be moved to MaterialXTest for now as scripts in that folder are not part of the distribution.

Reply:
I've revamped the script so that it can handle generating material graphs from any file which has definitions.
So the TestSuite/stdlib/surface.mtlx which has a checker definition can have a graph generated. See updated check-in comment.